src/Stratis.Bitcoin.Features.PoA/PoAMiner.cs

```diff
@@ -1,16 +1,121 @@
 ﻿using System;
-using System.Collections.Generic;
-using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Consensus;
+using Stratis.Bitcoin.Interfaces;
+using Stratis.Bitcoin.Mining;
+using Stratis.Bitcoin.Utilities;
 
 namespace Stratis.Bitcoin.Features.PoA
 {
-    public class PoAMiner
+    public class PoAMiner : IDisposable
```

Comment:
Can we have an interface so that we have `PoaMining : IPoaMining, Idisposable` (like in IPosMinting and IPowMining)?

Reply:
Yeah. Later when we work on production version of miner we shouldn't forget about adding the interface and commenting it properly.

Reply:
I really think adding the interface now is a good move. When you want to test etc, you have to have an interface. If it changes, than it's very quick and easy to change.

Reply:
ok, fixed